### DISCUSSION TITLE
Accessing the dockableWindows

### DISCUSSION THREAD

**Discussion by algotron** [pthom/hello_imgui#116] [Q&A] [answered]
I would like to access the dockableWindows that are defined in the HelloImGui::AbstractRunner.  AbstractRunner contains the HelloImGui::DockingParams, which contains std::vector<HelloImGui::DockableWindow> could be accessed from gLastRunner in the code below (taken from hello_imgui.cpp), but I cannot access gLastRunner.

Here is the code where gLastRunner is defined in hello_imgui.cpp:

void Run(RunnerParams& runnerParams)
{
    IM_ASSERT(_CheckAdditionLayoutNamesUniqueness(runnerParams));
    gLastRunner = FactorRunner(runnerParams);
    if (gLastRunner == nullptr)
    {
        fprintf(stderr, "HelloImGui::Run() failed to factor a runner!\n %s", gMissingBackendErrorMessage.c_str());
        IM_ASSERT(false && "HelloImGui::Run() failed to factor a runner!");
    }
    gLastRunnerParams = &runnerParams;
    gLastRunner->Run();
}

I have been thinking along the lines of something like this to access the dockableWindows:

 HelloImGui::RunnerParams & _runner_params = gLastRunner;
  HelloImGui::DockingParams & _dockingParams = _runner_params.dockingParams;
  std::vector<HelloImGui::DockableWindow> & _dockable_windows = _dockingParams.dockableWindows;

But there is no apparent way to access gLastRunner.

Are the dockableWindows accessible? and if so, how may I access them?

Thank-you.

**Comment by pthom** [comment, marked as answer]
You can query 


RunnerParams* HelloImGui::GetRunnerParams();

Is this what you are looking for ?

**Reply by algotron** [reply]
Yes, thank-you very much for the prompt reply.

This works well:
HelloImGui::RunnerParams * _cprunner_params = HelloImGui::GetRunnerParams();
  HelloImGui::DockingParams & _dockingParams = _cprunner_params->dockingParams;
  std::vector<HelloImGui::DockableWindow> & _dockable_windows = _dockingParams.dockableWindows;

**Comment by algotron** [comment]
Yes, thank-you Pascal.

Yours truly,

Peter Walker
